AppController/lib/cron_helper.rb

```diff
@@ -285,35 +285,147 @@ def self.fix_months(months)
 
     if splitted.length == 3
       months_of_year = "every"
       time = splitted[2]
-    else
+      hour, min = time.split(":")
+    elsif splitted.length == 5
       months_of_year = splitted[3]
       time = splitted[4]
+      hour, min = time.split(":")
+    else    # schedule length = 7, e.g. every 7 minutes from 10:00 to 14:00
+      months_of_year = "every"
+      days_of_week = "day"
+
+      increment = splitted[1].to_i
+      increment_type = splitted[2]
+
+      # Split hour and minute and trim leading zeros.
+      # Start time is represented as t1 = h1:m1.
+      # End time is represented as t2 = h2:m2.
+      h1, m1 = splitted[4].split(":").map(&:to_i)
+      h2, m2 = splitted[6].split(":").map(&:to_i)
+
+      if h1 == h2 && m1 == m2   # e.g. every 5 minutes from 12:05 to 12:05
+        return []
+      elsif increment_type == "hours"
+        if increment > 24       # Invalid increment for this type of schedule.
+          return []
+        elsif increment == 24   # e.g. every 24 hours from 01:30 to 01:29
+          hour = h1
+          min = m1
+        elsif h1 < h2
+          if m1 <= m2   # hours, h1 < h2, m1 <= m2
+            hour = "#{h1}-#{h2}/#{increment}"
+            min = "#{m1}"
+          else          # hours, h1 < h2, m1 > m2
+            hour = "#{h1}-#{h2-1}/#{increment}"
+            min = "#{m1}"
+          end
+        elsif h1 >= h2   # hours, h1 >= h2
+          multiple_cron_entries = true
+
+          # Batch 1 - before midnight
+          hour = "#{h1}-23/#{increment}"
+          min = "#{m1}"
+          crons.push({"hour" => hour, "min" => min})
+
+          # Batch 2 - after midnight
+          remainder = (24 - h1) % increment
+          first_hour = (increment-remainder)
+          min = "#{m1}"
+          if m1 < m2    # hours, h1 >= h2, m1 < m2
+            last_hour = h2
+          else          # hours, h1 >= h2, m1 >= m2
+            last_hour = h2-1
+          end
+
+          # If the next occurrence after midnight is past h2,
+          # there won't be a batch 2.
+          if first_hour < last_hour
+            hour = "#{first_hour}-#{last_hour}/#{increment}"
+            crons.push({"hour" => hour, "min" => min})
+          elsif first_hour == last_hour
+            hour = "#{first_hour}"
+            crons.push({"hour" => hour, "min" => min})
+          end
+        end
+      else    # increment_type == minutes
+        multiple_cron_entries = true
+        first_of_hour = m1    # First occurrence of the hour.
+        if h1 < h2            # minutes, h1 < h2
+          for h in (h1..h2)
+            remainder  = (60 - first_of_hour) % increment
+            if h == h2
+              last_of_hour = m2
+            else
+              last_of_hour = 60 - remainder
+            end
+            mins = ""
+            for i in (first_of_hour..last_of_hour).step(increment)
+              mins += "#{i},"
+            end
+            crons.push({"hour" => "#{h}", "min" => mins[0..-2]})
```

Comment:
I think it would be clearer if you joined the array `(first_of_hour..last_of_hour).step(increment).to_a.join(',')`. It took me a bit to figure out why the `-2` was needed.